tests/console/libgcrypt.pm

```diff
@@ -0,0 +1,47 @@
+# SUSE's openQA tests
+#
+# Copyright © 2019 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: test that libgcrypt works as intended by executing
+#          the selftest and by calling libgcrypt-config.
+#          In addition to this test, also the gpg test should be
+#          checked by tester because gpg uses libgcrypt.
+# - execute libgcrypt self test
+# - using libgcrypt-config, check some parameters
+# Maintainer: Paolo Stivanin <pstivanin@suse.com>
+
+use base "opensusebasetest";
+use testapi;
+use utils;
+use strict;
+use warnings;
+use version_utils qw(is_sle is_opensuse);
+use registration;
+
+sub run {
+    select_console 'root-console';
+
+    assert_script_run "rpm -q libgcrypt20";
+    if (script_run("rpm -q libgcrypt-devel") == 1) {
+        zypper_call "in gcc libgcrypt-devel" if is_sle(">=12-SP4") || is_opensuse;
+    }
+
+    if (is_sle(">=12-SP4")) {
+        assert_script_run("curl -O " . data_url("libgcrypt-selftest.c"));
```

Comment:
You do not need to download the file, it is already present because of *prepare_test_data* testsuite. It is located in `~/data/libgcrypt-selftest.c`.

Reply:
didn't know that, thanks :)